### PR TITLE
tigervnc: unbreak the build (and add tests)

### PR DESCRIPTION
**Pull request by iblech**
###### Motivation for this change

Right now, the vncserver of tigervnc doesn't start because `xauth` is missing from `$PATH`. This pull request fixes this and also adds a basic test, both for tigervnc and for tightvnc.

Note that tightvnc in principle has the same xauth problem, but there it has already been solved by patching the start script (by replacing `xauth` with `${xauth}`). Here we simply add it to the path with `makeBinPath`. I don't know which procedure is best practice.

I was not able to actually run the test I added since the computers I have access to lack kvm support.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by iblech** [comment]
@GrahamcOfBorg test vnc

(I don't actually know whether I have permissions to ask our bot for this favor, but thought that I might just try it.)

**Comment by Mic92** [comment]
@GrahamcOfBorg test vnc

**Comment by jonringer** [comment]
@GrahamcOfBorg test vnc

File: nixos/tests/tigervnc.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      client.wait_for_window(r"VNC")
+      client.screenshot("screenshot")
+
+      server.succeed("vncserver -kill :1")
+    '';
+  }) { inherit system; });
+
+in with pkgs; {
+  testTigerVNC = makeVNCTest "tigervnc" tigervnc;
+  testTightVNC = makeVNCTest "tightvnc" tightvnc;
```
Comment (Mic92) [inline nixos/tests/tigervnc.nix:49]: Ok. Maybe the renaming was not the best idea. Just spotted that it tests two different implementations. However trying to build this, still causes infinite recursion.

Comment (iblech) [reply nixos/tests/tigervnc.nix:49]: Yes, the idea was to have one test file for both packages, as they are very similar in their functionality.

What's the best way to proceed? If I'm honest then I'm not completely sure why we use `passthru.tests.tigervnc = nixosTests.tigervnc` which I believe to cause the infinite recursion. I loosely based the test on the test for Mumble, and the Mumble package also does not contain an analogue of this declaration.

Thank you tending to this pull request! :-)

Comment (Mic92) [reply nixos/tests/tigervnc.nix:49]: Actually I use the same thing here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/networking/ferm/default.nix#L30

**Comment by Mic92** [comment]
I cherry-picked the vncserver fix itself into 4fb8fab4d6949fd1f499d0e5a312f8827dadda98
and backported it to 20.03 in 0321ceaf44fb4cee884f99a0f514991b363db5aa

**Comment by Ekleog** [comment]
@iblech AFAIU @Mic92 merged the fix, meaning that this PR only adds tests now. Are you still interested in rebasing on a recent master and pushing this to completion?

On that topic @Mic92, I can't see in the comments any reason for you not to merge the tests too. Or maybe you just didn't review them?

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**